Encrypted Ink/Services/MetamaskImporter.swift

```diff
@@ -0,0 +1,121 @@
+// Copyright © 2021 Encrypted Ink. All rights reserved.
+
+import SwiftStore
+import CryptoSwift
+import WalletCore
+
+class MetamaskImporter {
+    
+    static func importFromPath(_ metamaskDir: String, passphrase: String) -> (privateKeys: [String], mnemonics: [String])? {
+        let store = SwiftStore(dirPath: metamaskDir)
+        guard
+            let storageString = store.findKeys(key: "").first,
+            let storageData = storageString.data(using: .utf8),
+            let storage = try? JSONDecoder().decode(MetamaskStorage.self, from: storageData),
+            let decryptedStorage = decrypt(data: storage.keyringController.vault.data,
+                                           iv: storage.keyringController.vault.iv,
+                                           salt: storage.keyringController.vault.salt,
+                                           password: passphrase),
+            let parsedDecryptedStorage = (try? JSONSerialization.jsonObject(with: decryptedStorage, options: []) as? [[String: Any]])
+        else {
+            return nil
+        }
+        
+        var exportedPrivateKeys = [String]()
+        var exportedMnemonics = [String]()
+        for account in parsedDecryptedStorage {
+            guard let type = account["type"] as? String else { continue }
+            if type == "Simple Key Pair" {
+                guard let privateKeys = account["data"] as? [String] else { continue }
+                exportedPrivateKeys.append(contentsOf: privateKeys)
+            } else if type == "HD Key Tree" {
+                guard
+                    let data = account["data"] as? [String: Any],
+                    let mnemonic = data["mnemonic"] as? String,
+                    let numberOfAccounts = data["numberOfAccounts"] as? Int,
+                    let hdPath = data["hdPath"] as? String
+                else {
+                    continue
+                }
+                exportedMnemonics.append(mnemonic)
+                let wallet = HDWallet(mnemonic: mnemonic, passphrase: "")
+                let firstKey = wallet.getKeyForCoin(coin: .ethereum).data.hexString
+                
+                for accountIndex in 0..<numberOfAccounts {
+                    let privateKey = wallet.getKey(coin: .ethereum, derivationPath: hdPath + "/\(accountIndex)").data.hexString
+                    if privateKey != firstKey {
```

Comment:
Don't compare private keys.
Simply don't derive key for default derivation path.